index-v4.yaml

```diff
@@ -63,5 +63,10 @@ distributions:
     distribution_cache: http://repositories.ros.org/rosdistro_cache/melodic-cache.yaml.gz
     distribution_status: active
     distribution_type: ros1
+  noetic:
+    distribution: [noetic/distribution.yaml]
+    distribution_cache: http://repositories.ros.org/rosdistro_cache/noetic-cache.yaml.gz
```

Comment:
How are these `cache` archives created? Who creates them?

<details><summary>rosinstall_generator needs them</summary>

```
$ ROSDISTRO_INDEX_URL=https://raw.githubusercontent.com/ros/rosdistro/f2d7b2f5e5b7f1e30e7e1fcd4368b03990f2e552/index-v4.yaml rosinstall_generator --upstream-development --rosdistro noetic --repos std_msgs
Traceback (most recent call last):
  File "/usr/bin/rosinstall_generator", line 11, in <module>
    load_entry_point('rosinstall-generator==0.1.16', 'console_scripts', 'rosinstall_generator')()
  File "/usr/lib/python3/dist-packages/rosinstall_generator/cli.py", line 160, in main
    upstream_version_tag=args.upstream, upstream_source_version=args.upstream_development)
  File "/usr/lib/python3/dist-packages/rosinstall_generator/generator.py", line 253, in generate_rosinstall
    repo_names, unknown_repo_names = _classify_repo_names(distro_name, repo_names)
  File "/usr/lib/python3/dist-packages/rosinstall_generator/generator.py", line 81, in _classify_repo_names
    wet_distro = get_wet_distro(distro_name)
  File "/usr/lib/python3/dist-packages/rosinstall_generator/generator.py", line 222, in get_wet_distro
    _wet_distro = _get_wet_distro(distro_name)
  File "/usr/lib/python3/dist-packages/rosinstall_generator/distro.py", line 46, in get_distro
    return get_cached_distribution(index, distro_name)
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 137, in get_cached_distribution
    cache = get_distribution_cache(index, dist_name)
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 177, in get_distribution_cache
    yaml_str = get_distribution_cache_string(index, dist_name)
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 164, in get_distribution_cache_string
    yaml_gz_str = load_url(url, skip_decode=True)
  File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url
    fh = urlopen(url, timeout=timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found (http://repositories.ros.org/rosdistro_cache/noetic-cache.yaml.gz)
```
</details>

Reply:
A job on build.ros.org which can only be generated once the distro has been created.

Reply:
[This job](http://build.ros.org/job/melodic_rosdistro-cache) ? If I follow, it's generated by [this](https://github.com/ros-infrastructure/ros_buildfarm/blob/bf6286c4275bfaf18c600b2de42f9ce169ddb94e/scripts/generate_all_jobs.py#L167`) and [this](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/scripts/misc/generate_rosdistro_cache_job.py), but what calls them in the first place? is there a job to call the job generation scripts? Is that something done manually?

Reply:
> This job ?

Correct.

> If I follow, it's generated by this and this, but what calls them in the first place? 

The first one has to be called manually - it will generate these administrative jobs for a distro as well as reconfigure jobs for each build file. If new build files are added later the generation has to be run manually again.

Any configuration "within" a build file like adding/updating/deleting jobs for new packages/repos happens automatically by the nightly running reconfigure jobs.

Reply:
> The first one has to be called manually

First meaning the first time a rosdistro is added, or first time a build farm is set up?

Reply:
Any time something is added to the `index[-v4].yaml` jobs must be generated manually.